avocado_vt/test.py

```diff
@@ -260,16 +260,16 @@ def runTest(self):
         The actual testing happens inside setUp stage, this only
         reports the correct results
         """
-        if self.__status == "PASS":
-            pass
-        elif isinstance(self.__status, error.TestNAError):
-            self.cancel(self.__status)
-        elif isinstance(self.__status, error.TestWarn):
-            self.log.warn(details)
-        elif isinstance(self.__status, error.TestFail):
-            self.fail(self.__status)
-        else:
-            raise self.__status
+        if self.__status != "PASS":
+            if isinstance(self.__status, error.TestNAError):
+                self.cancel(str(self.__status))
+            elif isinstance(self.__status, error.TestWarn):
+                self.log.warn(str(self.__status))
+            elif isinstance(self.__status, error.TestFail):
+                self.fail(str(self.__status))
+            else:
+                raise ValueError('Unexpected test status: %s' %
+                                 self.__status)
```

Comment:
So this is the place where we have the fixes in a critical code path for the plugin. Will report back on the test results.